Code/Framework/AzCore/AzCore/IO/Streamer/RequestPath.h

```diff
@@ -24,33 +25,35 @@ namespace AZ
         {
         public:
             RequestPath() = default;
-            RequestPath(const RequestPath& rhs);
-            RequestPath(RequestPath&& rhs);
+            RequestPath(const RequestPath& rhs) = default;
+            RequestPath(RequestPath&& rhs) = default;
+            // Allow path views to be cast to a request path.
+            RequestPath(AZ::IO::PathView path);
 
-            RequestPath& operator=(const RequestPath& rhs);
-            RequestPath& operator=(RequestPath&& rhs);
+            RequestPath& operator=(const RequestPath& rhs) = default;
+            RequestPath& operator=(RequestPath&& rhs) = default;
+            RequestPath& operator=(AZ::IO::PathView path);
 
             bool operator==(const RequestPath& rhs) const;
             bool operator!=(const RequestPath& rhs) const;
 
-            void InitFromRelativePath(AZStd::string path);
-            void InitFromAbsolutePath(AZStd::string path);
-
             bool IsValid() const;
             void Clear();
 
-            const char* GetAbsolutePath() const;
-            const char* GetRelativePath() const;
+            const char* GetAbsolutePathString() const;
+            const char* GetRelativePathString() const;
```

Comment:
Could we call rename this function to GetAbsolutePathCStr() and GetRelativePathCStr() to help prevent confusion at first glance that this function might be returning a string object?